### PR TITLE
SNOW-860063 Update documentation

### PR DESCRIPTION
**Pull request by sfc-gh-stan**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #NNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: src/snowflake/snowpark/functions.py [hunk 1]
```diff
@@ -7034,6 +7034,29 @@ def pandas_udtf(
...
+    and returning results as
+    `Pandas DataFrames <https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html>`_
+    or lists of `Pandas arrays <https://pandas.pydata.org/docs/reference/api/pandas.array.html>`_
+    or `Pandas Series <https://pandas.pydata.org/docs/reference/series.html>`_.
+    In addition, vectorized Python UDTFs allow for easy integration with libraries that operate on pandas DataFrames or pandas arrays.
+
+    A vectorized UDTF handler class:
+    - defines an :code:`end_partition` method that takes in a DataFrame argument and returns a :code:`pandas.DataFrame` or a tuple of :code:`pandas.Series` or :code:`pandas.arrays` where each array is a column.
+    - does NOT defines a :code:`process` method.
+    - optionally defines a handler class with an :code:`__init__` method which will be invoked before processing each partition.
```
Comment (sfc-gh-aalam) [inline src/snowflake/snowpark/functions.py:7050]:
```suggestion
    - does NOT define a :code:`process` method.
    - optionally defines a handler class with an :code:`__init__` method which will be invoked before processing each partition.
```

Comment (sfc-gh-aalam) [reply src/snowflake/snowpark/functions.py:7050]: same comment below

File: src/snowflake/snowpark/udtf.py [hunk 2]
```diff
@@ -299,15 +299,29 @@ class UDTFRegistration:
...
+    and returning results as
+    `Pandas DataFrames <https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html>`_
+    or lists of `Pandas arrays <https://pandas.pydata.org/docs/reference/api/pandas.array.html>`_
+    or `Pandas Series <https://pandas.pydata.org/docs/reference/series.html>`_.
+    Vectorized Python UDTFs allow for easy integration with libraries that operate on pandas DataFrames or pandas arrays.
+
+    A vectorized UDTF handler class:
+    - defines an :code:`end_partition` method that takes in a DataFrame argument and returns a :code:`pandas.DataFrame` or a tuple of :code:`pandas.Series` or :code:`pandas.arrays` where each array is a column.
+    - does NOT defines a :code:`process` method.
+    - optionally defines a handler class with an :code:`__init__` method which will be invoked before processing each partition.
```
Comment (sfc-gh-aalam) [inline src/snowflake/snowpark/udtf.py:314]:
```suggestion
    - does NOT define a :code:`process` method.
    - optionally defines a handler class with an :code:`__init__` method which will be invoked before processing each partition.
```

File: src/snowflake/snowpark/functions.py [hunk 3]
```diff
@@ -7034,6 +7034,29 @@ def pandas_udtf(
         - :func:`udtf`
         - :meth:`UDTFRegistration.register() <snowflake.snowpark.udf.UDTFRegistration.register>`
 
+    Compared to the default row-by-row processing pattern of a normal UDTF, which sometimes is
+    inefficient, vectorized Python UDTFs (user-defined table functions) enable seamless partition-by-partition processing
```
Comment (sfc-gh-sfan) [inline src/snowflake/snowpark/functions.py:7038]: We probably should move the explanation of UDTF to the beginning of the doc?

File: src/snowflake/snowpark/functions.py [hunk 4]
```diff
@@ -7034,6 +7034,29 @@ def pandas_udtf(
...
+    `Pandas DataFrames <https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html>`_
+    and returning results as
+    `Pandas DataFrames <https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html>`_
+    or lists of `Pandas arrays <https://pandas.pydata.org/docs/reference/api/pandas.array.html>`_
+    or `Pandas Series <https://pandas.pydata.org/docs/reference/series.html>`_.
+    In addition, vectorized Python UDTFs allow for easy integration with libraries that operate on pandas DataFrames or pandas arrays.
+
+    A vectorized UDTF handler class:
+    - defines an :code:`end_partition` method that takes in a DataFrame argument and returns a :code:`pandas.DataFrame` or a tuple of :code:`pandas.Series` or :code:`pandas.arrays` where each array is a column.
+    - does NOT defines a :code:`process` method.
```
Comment (sfc-gh-sfan) [inline src/snowflake/snowpark/functions.py:7049]: Is this enforced in our code as well?

Comment (sfc-gh-sfan) [reply src/snowflake/snowpark/functions.py]: If it is enforced on the backend then it's okay. Ideally it is better to fail earlier but if that requires a lot of changes on the client then we could just rely on the backend.

**Review by sfc-gh-sfan — APPROVED** [review]

File: src/snowflake/snowpark/functions.py (hunk 4 above)
Comment (sfc-gh-stan) [reply src/snowflake/snowpark/functions.py]: It should not require too many changes. I can include it in the next release. https://snowflakecomputing.atlassian.net/browse/SNOW-879777

**Review by sfc-gh-mabrennan — APPROVED** [review]
Docstrings look great. I made a couple of suggestions. Thanks!

**Pull request merged**